map/mwm_url.cpp

```diff
@@ -163,17 +229,33 @@ ParsedMapApi::ParsingResult ParsedMapApi::Parse(Uri const & uri)
       return ParsingResult::Route;
     }
     case ApiURLType::Search:
+    {
       SearchRequest request;
       if (!uri.ForEachKeyValue(bind(&ParsedMapApi::SearchKeyValue, this, _1, _2, ref(request))))
         return ParsingResult::Incorrect;
       
       m_request = request;
       return request.m_query.empty() ? ParsingResult::Incorrect : ParsingResult::Search;
+    }
+    case ApiURLType::Lead:
+    {
+      lead::CompaignDesription description;
+      if (!uri.ForEachKeyValue(bind(&ParsedMapApi::LeadKeyValue, this, _1, _2, ref(description))))
```

Comment:
We've already discussed it before. And you decided this is so non-trivial to use lambda here.